### ISSUE TITLE
Unit tests that depend on the compiler emitting specific code could fail in case the compiler is updated

### ISSUE THREAD

**Issue by zsr2531** [Washi1337/Echo#56]
**Summary**: Unit tests like [this](https://github.com/Washi1337/Echo/blob/d3815b205efbd50f816facfb2ba0106aa5ebe2c4/test/Platforms/Echo.Platforms.AsmResolver.Tests/StateTransitionResolverTest.cs#L32) depend on the compiler emitting a specific sequence of IL.

In the case of that specific unit test, the expected IL is something like this:
```
ldc.i4.s 18
bge adult

ldstr "Child"
br end

adult:
ldstr "Adult"

end:
ret
```

However, I have just found that the compiler emitted something like this:

```
ldc.i4.s 18
bge adult

ldstr "Child"
ret

adult:
ldstr "Adult"
ret
```

Thus, the unit test fails, as it expects to find only 1 `ret` instruction at the end of the method, with both branches reaching it.

**Possible solution**: Use a tool like [InlineIL.Fody](https://github.com/ltrzesniewski/InlineIL.Fody) to fine tune what IL gets emitted.

**Comment by HoLLy-HaCKeR** [comment]
I'll take a look at this

**Comment by zsr2531** [comment]
@HoLLy-HaCKeR Want me to assign this issue to you? Or are you still working on it?

**Comment by HoLLy-HaCKeR** [comment]
I havent been programming much lately, so if this is urgent, feel free to take over.

**Comment by zsr2531** [comment]
There is no such hurry, take your time :)

**Comment by Washi1337** [comment]
Implemented using #59
